Dockerfile

```diff
@@ -186,12 +149,23 @@ RUN make entrypoints
 RUN --mount=type=cache,target=/root/.cache \
     --mount=type=cache,target=/var/lib/localstack/cache \
     source .venv/bin/activate && \
+    python -m localstack.cli.lpm install java --version 11 && \
     python -m localstack.cli.lpm install \
       lambda-runtime \
       dynamodb-local && \
     chown -R localstack:localstack /usr/lib/localstack && \
     chmod -R 777 /usr/lib/localstack
 
+# Set up Java
+ENV JAVA_HOME /usr/lib/localstack/java/11
+RUN { \
+        echo '#!/bin/sh'; echo 'set -e'; echo; \
+        echo 'dirname "$(dirname "$(readlink -f "$(which javac || which java)")")"'; \
+    } > /usr/local/bin/docker-java-home \
+    && chmod +x /usr/local/bin/docker-java-home
+RUN ln -s $JAVA_HOME/bin/java /usr/bin/java
+ENV PATH="${PATH}:${JAVA_HOME}/bin"
```

Comment:
q: Is there a chance to completely avoid setting Java home globally? We want to be able to switch between versions, and this could be a pitfall in such a case...
This would mean that, when using Java, you would have to get `JAVA_HOME` from the package first and set it in the environment when calling the `java` command (which would make it explicit that it can vary and there are actually only very few usages of Java).

Reply:
Yes, this should be possible. But first we need to update all the Java components with explicit installer dependencies and make them set the relevant `JAVA_HOME`. This is planned in a follow up iteration.